File: sangsoo/0X1C Floyd/11404.c
#include <stdio.h>
#include <string.h>

#define INF 0x3f3f3f3f
#define MAXN 105

int d[MAXN][MAXN];
int n, m;

int min(int a, int b) {
    return a < b ? a : b;
}

int main() {
    scanf("%d %d", &n, &m);
    
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            d[i][j] = INF;
        }
    }
    
    for (int i = 0; i < m; i++) {
        int a, b, c;
        scanf("%d %d %d", &a, &b, &c);
        d[a][b] = min(d[a][b], c);  // Considering multiple routes.
    }
    
    for (int i = 1; i <= n; i++) d[i][i] = 0;

    for (int k = 1; k <= n; k++) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                d[i][j] = min(d[i][j], d[i][k] + d[k][j]);
            }
        }
    }

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            if (d[i][j] == INF) {
                printf("0 ");
            } else {
                printf("%d ", d[i][j]);
            }
        }
        printf("\n");
    }

    return 0;
}

File: sangsoo/0X1C Floyd/11404.py
from sys import stdin
input = stdin.readline

INF = float('inf')

n = int(input())
m = int(input())

d = [[INF for _ in range(n+1)] for _ in range(n+1)]

for _ in range(m):
    a, b, c = map(int, input().split())
    d[a][b] = min(d[a][b], c)  # Considering multiple routes.

for i in range(1, n+1):
    d[i][i] = 0

for k in range(1, n+1):
    for i in range(1, n+1):
        for j in range(1, n+1):
            d[i][j] = min(d[i][j], d[i][k] + d[k][j])

for i in range(1, n+1):
    for j in range(1, n+1):
        if d[i][j] == INF:
            print(0, end=" ")
        else:
            print(d[i][j], end=" ")
    print()
